nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/test/CoreOperationsTest.java

```diff
@@ -650,6 +653,25 @@ public void testCreateVersion3() throws Exception {
         }
     }
 
+    @Test
+    public void testCheckInVersion() throws Exception {
+        testCheckInVersion("none", "0.0");
+        testCheckInVersion("minor", "0.1");
+        testCheckInVersion("major", "1.0");
+        testCheckInVersion("anything_not_in_CheckInDocument.options", "0.1");
+    }
+
+    protected void testCheckInVersion(String versioningOption, String expectedVersion) throws Exception {
+        DocumentModel docVersion = session.createDocumentModel("/src", "versionMe", "File");
```

Comment:
As this is called 4 times, you're attempting to creating 4 documents with the same name in the same parent. We try to detect collisions and use a different name but it would be better to directly use a unique name instead, for clarity. For instance `"versionMe-" + versioningOption`.

Or split the `testCheckInVersion()` test method into 4 different test methods, each doing one test call.